File: template/src/app/pages/admin/menu/model/MenuItemDto.ts
import { numeric, NumericValueType, prop, required } from '@rxweb/reactive-form-validators';

export class MenuItemDto {
    @prop()
    id?: number;
    item?: string;
    @required({ message: 'pages.modal_user.validation_required' })
    label?: string;
    @prop()
    description?: string;
    @prop()
    parent?: string;
    @prop()
    @numeric({ allowDecimal: false, acceptValue: NumericValueType.PositiveNumber })
    visualOrder: number = 0;
    @prop()
    uri?: string;
    @prop()
    icon?: string;
    realm?: string;
    @prop()
    parentId?: string;
    @required({ message: 'pages.modal_user.validation_required' })
    roleId?: string;
    @prop()
    roleName?: string;
    @required({ message: 'pages.modal_user.validation_required' })
    visible?: boolean;
}

File: mim-dto/src/main/java/it/atlantica/dto/MenuItemDto.java
package it.atlantica.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class MenuItemDto {
    private Long id;
    private String item;
    @NotNull
    @NotEmpty
    @NotBlank
    private String label;

    private String description;
    private String parent;
    private Long parentId;
    private int visualOrder = 0;
    private String uri;
    private String icon;
    private String realm;
    @NotNull
    @NotEmpty
    @NotBlank
    private String roleId;
    @NotNull
    private boolean visible;

    private String roleName;
}